src/state/slices/opportunitiesSlice/resolvers/idle.ts

```diff
@@ -0,0 +1,100 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { fromAssetId, toAssetId } from '@shapeshiftoss/caip'
+import { DefiProvider, DefiType } from 'features/defi/contexts/DefiManagerProvider/DefiCommon'
+import { getIdleInvestor } from 'features/defi/contexts/IdleProvider/idleInvestorSingleton'
+import { logger } from 'lib/logger'
+import { selectAssetById } from 'state/slices/selectors'
+
+import type {
+  GetOpportunityIdsOutput,
+  GetOpportunityMetadataOutput,
+  OpportunitiesState,
+  OpportunityDefiType,
+  StakingId,
+} from '../types'
+import type { ReduxApi } from './types'
+
+const moduleLogger = logger.child({ namespace: ['opportunities', 'resolvers', 'idle'] })
+
+export const idleStakingOpportunitiesMetadataResolver = async ({
+  opportunityType,
+  reduxApi,
+}: {
+  opportunityType: OpportunityDefiType
+  reduxApi: ReduxApi
+}): Promise<{ data: GetOpportunityMetadataOutput }> => {
+  const idleInvestor = getIdleInvestor()
+  const opportunities = await idleInvestor.findAll()
+
+  const { getState } = reduxApi
+  const state: any = getState() // ReduxState causes circular dependency
+
+  const byId = await opportunities.reduce(
```

Comment:
```suggestion
  const metaDataById = await opportunities.reduce(
```

Or something more descriptive.